manifests/config.pp

```diff
@@ -21,12 +22,24 @@
     warning("${module_name}: The \$merge_options parameter will default to true in the next major release. Please review the documentation regarding the implications.")
   }
 
-  if $config_dir != undef {
-    file { $config_dir:
-      ensure => directory,
-      owner  => 'root',
-      group  => 'root',
-      mode   => '0755',
+  # $config_dir should only be created if default value is used and the OS
+  # is Gentoo or if a custom value is used
+  if $config_dir {
+    if $config_dir == $haproxy::config_dir and $::osfamily == 'Gentoo' {
```

Comment:
Shouldn't this comparison be `if $config_dir == $haproxy::params::config_dir ...`? The value of `$config_dir` in this manifest is passed down from `haproxy` (init.pp), through `haproxy::instance`. The values of `$config_dir` and `$haproxy::config_dir` will always be the same, no?